Added the propsed html structure taken from Wordpress

ARBaynes opened the pull request.

ARBaynes committed:
Added the propsed html structure taken from Wordpress

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+	function set_taxonomies ($post) {
+		$this->taxonomies = get_the_taxonomies($post);
+	}
+	//only run after set_taxonomies()
```
Is this comment still necessary, given that you're performing a test within the function? It might be a bit confusing later on.

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+	function set_url ($post) {
+		$this->url = get_permalink($post);
+	}
+	function set_title ($post) {
```
This is a minor point but this file has inconsistent line spacing. In some places there is a blank line between functons, but not in others. It's probably best to be consistent (and I'd suggest having the blank line between, just so it's easier to scan)

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		}  else {
+			$this->categories = strip_tags($this->taxonomies['category']);
```
I'm not sure if the `$this->categories` assignment should be in the `else` here. Would it still work as expected if there was no `else` and the `$this->categories` always occured?

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		} else {
+			$this->tags = strip_tags( $this->taxonomies['post_tag'] );
```
As with the `$this->categories` comment above, I'm not sure if the `$this->tags` assignment should be in the `else` here. Would it still work as expected if there was no `else` and the `$this->tags` always occured?

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		} else {
+			$this->guidance = strip_tags($this->taxonomies['guidance']);
```
See comments above for the `$this->categories` and `$this->tags` assignments. Is the same true here?

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+	}
+	//can only run after set_guidance()
+	function set_recommended () {
+		if (!strpos($this->guidance, 'Recommended')) {
```
This is up to you, but this is a really long file and I wonder if you might want to use the ternary operator here. It's a perfect candidate for ternary (simple assignment based on outcome of a condition), and it would help you avoid the slightly confusing 'if this is not' condition

```php
$this->recommended = (strpos($this->guidance, 'Recommended')) ? 'true' : 'false';
```

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+		}
+	}
+	function set_online () {
+			if (!strpos($this->guidance, 'All online')) {
```
This is somewhere I'd consider using a ternary.

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+	}
+	//can only run after set_categories()
+	function set_partners () {
+		if (strpos($this->categories, 'Findmypast.co.uk') !== false || strpos($this->categories, 'Ancestry.co.uk') !== false ) {
```
I'm not sure this is an ideal candidate for an `if` condition. I'd suggest having an array of partners and iterating over them to identify whether they're a sub-string of `$this->categories`. If any partner is found set `$this->partners` to a string representing the name of the partner (see proposed-html-structure.html). Because you'll need both the name as it's stored in WordPress (i.e. Findmypast.co.uk) and the class name (i.e. 'find-my-past') this is a candidate for an associative array. Let me know if you need any help with this.

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+		}
+	}
+
+	function get_url () {return $this->url;}
```
If these aren't being used I'd either delete them or replace the `sprintf()` that's used to create `$compiled` with calls to the getters. Personally, I'd probably just delete them because they don't need to be exposed to other objects.

gtvj commented on proposed-html-structure-from-wordpress.php:
```diff
@@ -0,0 +1,185 @@
+	$research_guide_array = array();
+
+	// The Loop
+	if ( $research_guide_query->have_posts() ) :
```
This alternative syntax for control structures is best used when you've got PHP interspersed within HTML (because it makes it clearer where the control structure ends, rather than a closing `}`). In this case I think you should use the normal syntax.

gtvj commented:
Hi @Urborging @Ash98 - you've done a really good job on this. I was impressed by how you've approached it (especially the use of classes, visibility and setter methods etc.). I've added quite a few comments in the code review which I hope are helpful. Nice work!

I'd like this PR to include the two HTML structures that are output too (since that what we'll be consuming in Angular). Pop over to my desk tomorrow and we'll talk about it :+1:

ARBaynes committed:
Suggested changes made to the proposed html structure taken from Wordpress

ARBaynes committed:
Now using slugs instead of names for taxonomy-based fields

ARBaynes committed:
Added related files for the proposed HTML structure (and folder to contain them)

ARBaynes committed:
Removed unused field 'taxonomies'

gtvj commented:
HI @Urborging @Ash98 - I've come across a small problem that I'd be grateful if you could fix. The HTML output from WordPress has a `data-guide-recommended-for-category` attribute which is currently either `'true'` or `'false'`. This isn't quite right because it needs to be either `'false'` or the name of the recommended category. Would you be able to fix this? I'm sure @netdesignr will be happy to help if you have any problems

ARBaynes committed:
'Recommended' should now return the recommended guides if not false

although I am not currently at a computer that can test this, in theory it should work.
